performance/neo4j-streams-pt.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import configparser
+import uuid
+import time
+import matplotlib.pyplot as plt
+
+from py2neo import Graph
+
+parser = argparse.ArgumentParser(description='Performance Test suite for neo4j-streams')
+
+parser.add_argument('--out', help='output file', dest='out')
+parser.add_argument('--start', nargs='*', metavar=('msg','repeats'), help='print the reference time (avg repeats * msg), in order to find the [unit] values')
+parser.add_argument('--baseline', nargs='+', metavar='pow', help='runs the [unit] * pow')
+
+args = parser.parse_args()
+
+config = configparser.ConfigParser()
+
+config.read('config.ini')
+
+producerConfig = config['producer']
+consumerConfig = config['consumer']
+unit = config['unit']
+
+producer = Graph(producerConfig['url'], user=producerConfig['username'], password=producerConfig['password'])
+consumer = Graph(consumerConfig['url'], user=consumerConfig['username'], password=consumerConfig['password'])
+
+def executeTest(nodes):
+	"""Returns elapsed per node """
+	test_ref = str(uuid.uuid1())
+
+	result = producer.run("""
+		UNWIND range(1,{msg}) as ran
+		CREATE (n:Performance)
+		SET n.group = {uuid}, n.creation_time = apoc.date.currentTimestamp()
+		RETURN min(n.creation_time) as creation_time
+		""", msg = nodes, uuid=test_ref).data()
+
+	creation_time = result[0]['creation_time']
+	#print(test_ref,creation_time)
+	received_time = creation_time
+
+	in_streaming = True
+	while (in_streaming):			
+		time.sleep(1)
+		
+		result = consumer.run("""
+			MATCH (n:Performance)
+			WHERE n.group = {uuid}
+			RETURN count(n) as cnt, max(n.received_time) as received_time
+			""", uuid=test_ref).data()
+
+		#print(result)
+		in_streaming = result[0]['cnt'] < nodes
+		received_time = result[0]['received_time']
+
+	return (received_time - creation_time) / nodes
```

Comment:
probably better to plot it not as percentage but use nodes as X axis and then the  time delta

Reply:
It's easier to compare the milliseconds per node instead of the whole elapsed time between very different series (ms vs sec or minutes).